linera-execution/src/system.rs

```diff
@@ -517,9 +517,31 @@ where
                 for (chain_id, stream_id, next_index) in streams {
                     let subscriptions = self
                         .event_subscriptions
-                        .get_mut_or_default(&(chain_id, stream_id))
+                        .get_mut_or_default(&(chain_id, stream_id.clone()))
+                        .await?;
+                    if subscriptions.next_index >= next_index {
+                        continue;
+                    }
+                    subscriptions.next_index = next_index;
+                    for application_id in &subscriptions.applications {
+                        txn_tracker.add_stream_to_process(
+                            *application_id,
+                            chain_id,
+                            stream_id.clone(),
+                            next_index,
+                        );
+                    }
+                    let index = next_index
+                        .checked_sub(1)
+                        .ok_or(ArithmeticError::Underflow)?;
+                    self.context()
+                        .extra()
+                        .get_event(EventId {
+                            chain_id,
+                            stream_id: stream_id.clone(),
+                            index,
+                        })
```

Comment:
This is just to make sure that we have that event locally?

Reply:
Yes. Event values are supposed to be small anyway, but I can replace this with something like `contains_event`.

Reply:
I was wondering whether this call here is needed at all? The assumption is that we have the events, it's not like we need to sync it here

Reply:
In the client maybe, but the server needs to verify that the events the block proposer claims exist actually exist.